libs/api-client-bear/src/utils/export.ts

```diff
@@ -1,7 +1,15 @@
-// (C) 2007-2019 GoodData Corporation
+// (C) 2007-2023 GoodData Corporation
 import { SUCCESS_REQUEST_STATUS, BAD_REQUEST_STATUS } from "../constants/errors";
 
-export function isExportFinished(responseHeaders: Response): boolean {
-    const taskState = responseHeaders.status;
+export function isExportFinished(response: Response): boolean {
+    const taskState = response.status;
     return taskState === SUCCESS_REQUEST_STATUS || taskState >= BAD_REQUEST_STATUS; // OK || ERROR
 }
+
+export const parseFileNameFromContentDisposition = (response: Response): string | undefined => {
+    const contentDispositionHeader = response.headers.get("Content-Disposition") || "";
+    // eslint-disable-next-line regexp/no-unused-capturing-group
+    const matches = /filename\*?=([^']*'')?([^;]*)/.exec(contentDispositionHeader);
+    const urlEncodedFileName = matches ? matches[2] : undefined;
+    return urlEncodedFileName ? decodeURIComponent(urlEncodedFileName) : undefined;
```

Comment:
Is undefined fallback correct here ? Should be there some generic name  like exportData or something ?

Reply:
If `undefined` is used, it will fallback on URI value in the downloader util, but I can use something generic. However, it will be missing the extension. I would leave it as is as it is unlikely exporter would stop sending `content-disposition` header with file name.